notebook/static/tree/js/notebooklist.js

```diff
@@ -697,6 +707,73 @@ define([
         });
     };
 
+    NotebookList.prototype.move_selected = function() {
+        // TODO(nhdaly): Support moving multiple items at once.
+        if (this.selected.length !== 1){
+            return;
+        }
+
+        var that = this;
+        var item_path = this.selected[0].path;
+        var item_name = this.selected[0].name;
+        var item_type = this.selected[0].type;
+
+        // Open a dialog to enter the new path, with current path as default.
+        var input = $('<input/>').attr('type','text').attr('size','25').addClass('form-control')
+            .val(utils.url_path_join('/', that.notebook_path));
+        var dialog_body = $('<div/>').append(
+            $("<p/>").addClass("rename-message")
+                .text('Enter new destination directory path for '+ item_type + ':')
+        ).append(
+            $("<br/>")
+        ).append(input);
+        var d = dialog.modal({
+            title : "Move "+ item_type,
+            body : dialog_body,
+            buttons : {
+                OK : {
+                    class: "btn-primary",
+                    click: function() {
+                      // Construct the new path using the user input and its name.
+                        var new_path = utils.url_path_join(input.val(), item_name)
+                        that.contents.rename(item_path, new_path).then(function() {
+                            that.load_list();
+                        }).catch(function(e) { 
+                            dialog.modal({
+                                title: "Move Failed",
+                                body: $('<div/>')
+                                    .text("An error occurred while moving \"" + item_name + "\" from \"" + item_path + "\" to \"" + new_path + "\".")
+                                    .append($('<div/>')
+                                        .addClass('alert alert-danger')
+                                        .text(e.message || e)),
+                                buttons: {
+                                    OK: {'class': 'btn-primary'}
+                                }
+                            });
+                            console.warn('Error durring moving :', e);
+                        });
+                    }
+                },
+                Cancel : {}
```

Comment:
We're looking to change our button orders so that the "Cancel" is always on the left and the action is on the right. So these should be switched to be compatible with future changes.